### PR TITLE
Workspace gl layers

### PR DESCRIPTION
**Pull request by nerik**
This adds the possibility to directly specify Mapbox GL JSON for workspace layers, allowing pulling layers from CARTO without having to have them specified in the `directory` endpoint nor in the base GL JSON style (use cases described here: https://github.com/GlobalFishingWatch/map-client/issues/932#issuecomment-412865322)

The gist of it is: 
- set up a CARTO table, have it public
- add the SQL query and a bunch of other parameters in the workspace layer `gl` parameter.
This is fully documented, with examples, here: https://github.com/GlobalFishingWatch/map-client/blob/a0e6283d539c75048a4cfa9221797f50d14b9b2b/documentation/workspaces.md
This is the workspace used for tests: https://github.com/GlobalFishingWatch/map-client/blob/a0e6283d539c75048a4cfa9221797f50d14b9b2b/public/workspace.json

@enriquetuya I made that choice because:
- this avoids having any conversion/parsing step on the client side code, reducing complexity (GL JSON is just taken from the workspace layer and merged with the base GL JSON style)
- it doesn't require creating a CARTO map/vizjson, only a public table and a SQL query are required.
- this gives complete control over the appeareance and behavior of the custom layer: SQL, popups, labels, various styling attributes, etc.
- from there, allowing custom raster and/or WMS workspace layers should be a breeze.

The disadvantage is that it is slightly more complex than it could be, so let me know if I should clarify anything.

### CODE REVIEW

File: public/workspace.json [hunk 1]
```diff
@@ -2,18 +2,11 @@
   "workspace": {
```
Comment (enriquetuya) [inline public/workspace.json]: @nerik are the changes in this file intended? maybe we want to have a `galery/` folder and include `custom-gl-layer-workspace.json`?

File: documentation/workspaces.md [hunk 2]
```diff
@@ -2,6 +2,118 @@
...
+  "title":	"South American countries",
+  "type": "CartoDBAnimation",
+  "id": "samerica_adm0",
+  "color": "#ff00ff",
+  "visible": true,
+  "gl": {
+    "source": {
+      "type": "vector",
+      "metadata": {
+        "gfw:carto-sql": "SELECT * FROM samerica_adm0",
```
Comment (enriquetuya) [inline documentation/workspaces.md]: @nerik we only support queries to our carto instance? where is that configured?

Comment (nerik) [reply documentation/workspaces.md]: @enriquetuya This is hardcoded here for now: https://github.com/GlobalFishingWatch/map-client/blob/develop/app/src/config.js#L205
Should I make this an env variable?

File: public/workspace.json (hunk 1 above)
Comment (nerik) [reply public/workspace.json]: @enriquetuya This is useful just for testing this PR. Maybe this shouldn't even be versioned? I like your idea, but that means maintaining those workspaces...

File: documentation/workspaces.md (hunk 2 above)
Comment (andres-arana) [reply documentation/workspaces.md]: +1 to making this configurable in an env file.

**Pull request merged**